pkg/api/v1/datafederation_types.go

```diff
@@ -0,0 +1,208 @@
+	PrivateEndpoints []DataFederationPE `json:"privateEndpoints,omitempty"`
+}
+
+type CloudProviderConfig struct {
+	AWS *AWSProviderConfig `json:"aws,omitempty"`
+}
+
+type AWSProviderConfig struct {
+	RoleID       string `json:"roleId,omitempty"`
+	TestS3Bucket string `json:"testS3Bucket,omitempty"`
+}
+
+type DataProcessRegion struct {
+	// +kubebuilder:validation:Enum:=AWS
+	CloudProvider string `json:"cloudProvider,omitempty"`
+	// +kubebuilder:validation:Enum:=SYDNEY_AUS;MUMBAI_IND;FRANKFURT_DEU;DUBLIN_IRL;LONDON_GBR;VIRGINIA_USA;OREGON_USA;SAOPAULO_BRA;SINGAPORE_SGP
+	Region string `json:"region,omitempty"`
+}
+
+type Storage struct {
+	Databases []Database `json:"databases,omitempty"`
+	Stores    []Store    `json:"stores,omitempty"`
+}
+
+type Database struct {
+	Collections            []Collection `json:"collections,omitempty"`
+	MaxWildcardCollections int          `json:"maxWildcardCollections,omitempty"`
+	Name                   string       `json:"name,omitempty"`
+	Views                  []View       `json:"views,omitempty"`
+}
+
+type Collection struct {
+	DataSources []DataSource `json:"dataSources,omitempty"`
+	Name        string       `json:"name,omitempty"`
+}
+
+type View struct {
+	Name     string `json:"name,omitempty"`
+	Pipeline string `json:"pipeline,omitempty"`
+	Source   string `json:"source,omitempty"`
+}
+
+type DataSource struct {
+	AllowInsecure   bool   `json:"allowInsecure,omitempty"`
+	Collection      string `json:"collection,omitempty"`
+	CollectionRegex string `json:"collectionRegex,omitempty"`
+	Database        string `json:"database,omitempty"`
+	DatabaseRegex   string `json:"databaseRegex,omitempty"`
+	// +kubebuilder:validation:Enum:=.avro;.avro.bz2;.avro.gz;.bson;.bson.bz2;.bson.gz;.bsonx;.csv;.csv.bz2;.csv.gz;.json;.json.bz2;.json.gz;.orc;.parquet;.tsv;.tsv.bz2;.tsv.gz
+	DefaultFormat       string   `json:"defaultFormat,omitempty"`
+	Path                string   `json:"path,omitempty"`
+	ProvenanceFieldName string   `json:"provenanceFieldName,omitempty"`
+	StoreName           string   `json:"storeName,omitempty"`
+	Urls                []string `json:"urls,omitempty"`
+}
+
+type Store struct {
+	Name                     string   `json:"name,omitempty"`
+	Provider                 string   `json:"provider,omitempty"`
+	AdditionalStorageClasses []string `json:"additionalStorageClasses,omitempty"`
+	Bucket                   string   `json:"bucket,omitempty"`
+	Delimiter                string   `json:"delimiter,omitempty"`
+	IncludeTags              bool     `json:"includeTags,omitempty"`
+	Prefix                   string   `json:"prefix,omitempty"`
+	Public                   bool     `json:"public,omitempty"`
+	Region                   string   `json:"region,omitempty"`
+}
+
+type DataFederationPE struct {
+	EndpointID string `json:"endpointId,omitempty"`
+	Provider   string `json:"provider,omitempty"`
+	Type       string `json:"type,omitempty"`
+}
+
+func (pe DataFederationPE) Identifier() interface{} {
```

Comment:
Why does it need to return `interface{}` instead of string?

Reply:
Saw later it's because of the `uitl.set` package, maybe worth to make the package generic?

Reply:
Yes, good idea. But better to do that in a separate PR.